homeassistant/components/unifiprotect/event.py

```diff
@@ -21,8 +21,28 @@
     EVENT_TYPE_FINGERPRINT_NOT_IDENTIFIED,
     EVENT_TYPE_NFC_SCANNED,
 )
-from .data import ProtectData, ProtectDeviceType, UFPConfigEntry
+from .data import (
+    Camera,
+    EventType,
+    ProtectAdoptableDeviceModel,
+    ProtectData,
+    ProtectDeviceType,
+    UFPConfigEntry,
+)
 from .entity import EventEntityMixin, ProtectDeviceEntity, ProtectEventMixin
+from .utils import Bootstrap
+
+
+def _getActiveUlpUser(bootstrap: Bootstrap, ulp_id: str) -> UlpUser | None:
+    """Get the active ULP user based on the event data."""
+    return next(
+        (
+            uu
+            for uu in bootstrap.ulp_users.values()
+            if uu.ulp_id == ulp_id and uu.status == "ACTIVE"
+        ),
+        None,
+    )
 
```

Comment:
Aren't these indexed by `id`? Can we avoid the linear search here?

If not we might want to go back and change the `uiprotect` code before we ship this while its likely no one is using this yet since it will be a breaking change to alter it later

Ideally we index by `ulp_id` and `registry_id` if thats isn't the same as `id` so we don't have a linear search here every time.